serial_server.py

```diff
@@ -108,21 +148,27 @@ def open(self, c, port=''):
         if not port:
             for i in range(len(self.SerialPorts)):
                 try:
-                    c['PortObject'] = Serial('\\\\.\\' + self.SerialPorts[i], timeout=0)
+                    c['PortObject'] = Serial(self.SerialPorts[i].devicepath, timeout=0)
                     break
                 except SerialException:
                     pass
             if 'PortObject' not in c:
                 raise NoPortsAvailableError()
         else:
-            try:
-                c['PortObject'] = Serial('\\\\.\\' + port, timeout=0)
-            except SerialException, e:
-                if e.message.find('cannot find') >= 0:
-                    raise Error(code=1, msg=e.message)
-                else:
-                    raise Error(code=2, msg=e.message)
-        return c['PortObject'].portstr.replace('\\\\.\\', '')
+            for x in self.SerialPorts:
+                if x.name.lower() == port.lower():
+                    try:
+                        c['PortObject'] = Serial(x.devicepath)
+                        # Strip off the leading \\.\ if it exists for
+                        # windows compatiblity
+                        return c['PortObject'].portstr.replace('\\\\.\\', '')
+                    except SerialException, e:
+                        if e.message.find('cannot find') >= 0:
```

Comment:
Ok, I see code 1 is used at the top of the file for `NoPortSelectedError`. If we want to make distinctions here, would probably be best to define more error subclasses like that.